scripts/dotnet-cli-build/SharedFrameworkTargets.cs

```diff
@@ -0,0 +1,155 @@
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Microsoft.Extensions.PlatformAbstractions;
+
+using static Microsoft.DotNet.Cli.Build.Framework.BuildHelpers;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class SharedFrameworkTargets
+    {
+        public const string SharedFrameworkName = "NETStandard.Library";
+
+        private const string CoreHostBaseName = "corehost";
+
+        [Target]
+        public static BuildTargetResult PublishSharedFramework(BuildTargetContext c)
+        {
+            string SharedFrameworkPublishRoot = Path.Combine(Dirs.Output, "obj", "sharedframework");
+            string SharedFrameworkSourceRoot = Path.Combine(Dirs.RepoRoot, "src", "sharedframework", "framework");
+            string SharedFrameworkNugetVersion = GetVersionFromProjectJson(Path.Combine(Path.Combine(Dirs.RepoRoot, "src", "sharedframework", "framework"), "project.json"));
+
+            if (Directory.Exists(SharedFrameworkPublishRoot))
+            {
+                Directory.Delete(SharedFrameworkPublishRoot, true);
+            }
+
+            // We publish to a sub folder of the PublishRoot so tools like heat and zip can generate folder structures easier.
+            string SharedFrameworkNameAndVersionRoot = Path.Combine(SharedFrameworkPublishRoot, "shared", SharedFrameworkName, SharedFrameworkNugetVersion);
+
+            DotNetCli.Stage0.Publish("--output", SharedFrameworkNameAndVersionRoot, SharedFrameworkSourceRoot).Execute().EnsureSuccessful();
+
+            c.BuildContext["SharedFrameworkPublishRoot"] = SharedFrameworkPublishRoot;
+            c.BuildContext["SharedFrameworkNugetVersion"] = SharedFrameworkNugetVersion;
+
+            // Clean up artifacts that dotnet-publish generates which we don't need
+            File.Delete(Path.Combine(SharedFrameworkNameAndVersionRoot, $"framework{Constants.ExeSuffix}"));
+            File.Delete(Path.Combine(SharedFrameworkNameAndVersionRoot, "framework.dll"));
+            File.Delete(Path.Combine(SharedFrameworkNameAndVersionRoot, "framework.pdb"));
+
+            // Rename the .deps file
+            File.Move(Path.Combine(SharedFrameworkNameAndVersionRoot, "framework.deps"), Path.Combine(SharedFrameworkNameAndVersionRoot, $"{SharedFrameworkName}.deps"));
+
+            // corehost will be renamed to dotnet at some point and then this can be removed.
+            File.Move(Path.Combine(SharedFrameworkNameAndVersionRoot, $"{CoreHostBaseName}{Constants.ExeSuffix}"), Path.Combine(SharedFrameworkNameAndVersionRoot, $"dotnet{Constants.ExeSuffix}"));
+
+            // hostpolicy will be renamed to dotnet at some point and then this can be removed.
+            File.Move(Path.Combine(SharedFrameworkNameAndVersionRoot, $"{Constants.DynamicLibPrefix}hostpolicy{Constants.DynamicLibSuffix}"), Path.Combine(SharedFrameworkNameAndVersionRoot, $"{Constants.DynamicLibPrefix}dotnethostimpl{Constants.DynamicLibSuffix}"));
+
```

Comment:
Has this `dotnethostxfr.dll` been added to the NuGet package yet? If so, which version has it?